File: Competitive Programming 3/Section 1.3.3/11687 - Digits/C++/main.cpp
#include <iostream>

using namespace std;

string line;

int countDigits(int number){
    int ans=0;
    for(;number;ans++,number/=10);
    return ans;
}

void countUntil1(int number){
    int ans = 0 ; 
    for(;number!=1;ans++,number = countDigits(number));
    cout << ans+2 << "\n";
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    while(getline(cin,line),line[0]!='E'){
        if(line.size()==1&&line[0]=='1'){
            cout << "1\n";
        }else{
            countUntil1(line.size());
        }
    }
    return 0;
}

File: Competitive Programming 3/Section 1.3.3/11687 - Digits/Python/main.py
from sys import stdin

def countDigits(number):
    ans = 0 
    while(number):
        ans+=1
        number//=10
    return ans
def countUntil1(number):
    ans = 0
    while number!=1:
        ans+=1
        number = countDigits(number)
    print(ans+2)

def main():
    for line in stdin:
        if line[0]=='E':
            break
        if line[0]=='1' and len(line.rstrip())==1:
            print(1)
        else:
            countUntil1(len(line.rstrip()))
main()